### PR TITLE
crimson/os/seastore/object_data_handler: cleanup read()

### PR DESCRIPTION
**Pull request by cyx1231st**
* Refactor based on the read range, pin range and extent range.
* MISC: Logs, variable names, asserts and indentions.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/crimson/os/seastore/object_data_handler.cc [hunk 1]
```diff
@@ -1441,93 +1441,116 @@ ObjectDataHandler::read_ret ObjectDataHandler::read(
...
+              DEBUGT("reading {}~{} from indirect-pin {}~{}, direct-pin {}~{}",
+                ctx.t,
+                l_current,
+                l_current_end - l_current,
+                pin_key,
+                pin_len,
+                e_key,
+                e_len);
+              assert(e_key <= pin_key);
+              assert(l_pin_end <= e_key + e_len);
```
Comment (cyx1231st) [inline src/crimson/os/seastore/object_data_handler.cc:1516]: Oops, clone exits in a different laddr range.

Comment (cyx1231st) [reply src/crimson/os/seastore/object_data_handler.cc]: There're other mistakes as well, fixed inplace.

**Comment by cyx1231st** [comment]
Should be ready for review.

**Review by myoungwon — APPROVED** [review]
I think this is a nice cleanup for readability.

**Review by xxhdx1985126 — APPROVED** [review]

**Pull request merged**